### PR TITLE
Test both negotiated & non-negotiated datachannels

### PR DESCRIPTION
**Pull request by jan-ivar**
Fixes `createDataChannelPair()` helper to test both negotiated and non-negotiated datachannels.

cc @alvestrand

### CODE REVIEW

**Comment by jan-ivar** [comment]
The [timeout](https://community-tc.services.mozilla.com/tasks/dAehhwd6RkWl4IjbMhuUIg/runs/0/logs/https%3A%2F%2Fcommunity-tc.services.mozilla.com%2Fapi%2Fqueue%2Fv1%2Ftask%2FdAehhwd6RkWl4IjbMhuUIg%2Fruns%2F0%2Fartifacts%2Fpublic%2Flogs%2Flive.log) looks unrelated.

File: webrtc/RTCDataChannel-bufferedAmount.html [hunk 1]
```diff
@@ -62,20 +62,19 @@
   0xe4, 0xb8, 0x96, 0xe7, 0x95, 0x8c,
   0xe4, 0xbd, 0xa0, 0xe5, 0xa5, 0xbd);
 
+for (const options of [{}, {negotiated: true, id: 0}]) {
```
Comment (alvestrand) [inline webrtc/RTCDataChannel-bufferedAmount.html:65]: The lack of indentation of the FOR loop makes my eyes hurt.
I appreciate the need to reduce the size of the delta, but please think of future readers.

File: webrtc/RTCDataChannel-bufferedAmount.html [hunk 2]
```diff
@@ -62,20 +62,19 @@
...
+
+const datachannel = `${options.negotiated? "negotiated " : ""}datachannel`;
```
Comment (alvestrand) [inline webrtc/RTCDataChannel-bufferedAmount.html:67]: Not the best of names. It's not a datachannel, it's a datachannel mode. What about calling it "mode"?

File: webrtc/RTCDataChannel-close.html [hunk 3]
```diff
@@ -8,10 +8,12 @@
 <script>
 'use strict';
 
+for (const options of [{}, {negotiated: true, id: 0}]) {
+
+const datachannel = `${options.negotiated? "negotiated " : ""}datachannel`;
```
Comment (alvestrand) [inline webrtc/RTCDataChannel-close.html:13]: Same comments as previous file.

File: webrtc/RTCDataChannel-send-blob-order.html [hunk 4]
```diff
@@ -5,13 +5,17 @@
 <script src="/resources/testharnessreport.js"></script>
 <script src="RTCPeerConnection-helper.js"></script>
 <script>
-promise_test(async t => {
+
+for (const options of [{}, {negotiated: true, id: 0}]) {
+  const datachannel = `${options.negotiated? "Negotiated d" : "D"}atachannel`;
```
Comment (alvestrand) [inline webrtc/RTCDataChannel-send-blob-order.html:10]: Here the indent is OK, at least.

**Review by alvestrand — APPROVED** [review]
Since https://wpt.fyi/results/webrtc/RTCDataChannel-close.html?sha=ee3f673ed8&label=pr_head now shows that the timeout hits non-negotiated datachannels only, I'm willing to accept this for expediency (one of my projects is stalled waiting for this fix), but would prefer that the name & indent issues be fixed before submitting.
Approving.

**Pull request merged**